commit 23a84627935c2cf880c0280d5a98a09e845dc126
Author: 7FM <41307817+7FM@users.noreply.github.com>
Date:   Sun Jan 19 13:37:39 2025 +0100

    Nix specific: do not check for updates

diff --git a/BridgeEmulator/services/updateManager.py b/BridgeEmulator/services/updateManager.py
index 560acc5..d814bf8 100644
--- a/BridgeEmulator/services/updateManager.py
+++ b/BridgeEmulator/services/updateManager.py
@@ -104,9 +104,4 @@ def githubInstall():
     pass
 
 def startupCheck():
-    if bridgeConfig["config"]["swupdate2"]["install"] == True:
-        bridgeConfig["config"]["swupdate2"]["install"] = False
-        bridgeConfig["config"]["swupdate2"]["lastchange"] = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S")
-        bridgeConfig["config"]["swupdate2"]["bridge"]["lastinstall"] = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S")
-    versionCheck()
-    githubCheck()
+    pass
